### ISSUE TITLE
Error: MinClusterCount must be less than or equal to MaxClusterCount - When increasing min_cluster_count param of warehouse

### ISSUE THREAD

**Issue by wpl-sf-nbirch** [Snowflake-Labs/terraform-provider-snowflake#2652]
### Terraform CLI and Provider Versions

0.87.2

### Terraform Configuration

```terraform
Increase min_cluster_count from 1 to 2.

Original:
resource "snowflake_warehouse" "reporting_wh" {
  name                                = "REPORTING_WH"
  auto_resume                         = true
  auto_suspend                        = 600
  comment                             = "Reporting Warehouse"
  enable_query_acceleration           = false
  initially_suspended                 = true
  max_cluster_count                   = 10
  min_cluster_count                   = 1
  scaling_policy                      = "STANDARD"
  warehouse_size                      = "SMALL"
  warehouse_type                      = "STANDARD"
  statement_timeout_in_seconds        = 7200
  max_concurrency_level               = 16
}

Updated:
resource "snowflake_warehouse" "reporting_wh" {
  name                                = "REPORTING_WH"
  auto_resume                         = true
  auto_suspend                        = 600
  comment                             = "Reporting Warehouse"
  enable_query_acceleration           = false
  initially_suspended                 = true
  max_cluster_count                   = 10
  min_cluster_count                   = 2
  scaling_policy                      = "STANDARD"
  warehouse_size                      = "SMALL"
  warehouse_type                      = "STANDARD"
  statement_timeout_in_seconds        = 7200
  max_concurrency_level               = 16
}
```

### Expected Behavior

The minimum cluster count should increase to 2 nodes and allow auto-scale up to 10 nodes.

### Actual Behavior

**Error: MinClusterCount must be less than or equal to MaxClusterCount**
with snowflake_warehouse.reporting_wh
on snowflake_warehouses.tf line 125, in resource "snowflake_warehouse" "reporting_wh":
resource "snowflake_warehouse" "reporting_wh" {

### Steps to Reproduce

1. `terraform apply`

### How much impact is this issue causing?

High

**Comment by wpl-sf-nbirch** [comment]
It seems setting the min_cluster_count and max_cluster_count to the same value 2 works, and then increasing the max_cluster_count back to 10 works, as a two step process.

**Comment by sfc-gh-asawicki** [comment]
Hey @wpl-sf-nbirch. Thanks for reporting the issue.

The problem is the incorrect validation on the SDK side if the max_cluster_count is missing (in your case, not being updated). This also explains why doing the two-step works. This is a relatively easy fix, I will add this today.

**Comment by sfc-gh-asawicki** [comment]
Hey @wpl-sf-nbirch. Fix released in v0.88.0.

Please check and close the issue if the problem is solved for you.

**Comment by wpl-sf-nbirch** [comment]
Increasing and decreasing snowflake_warehouse min_cluster_count is now working without having to also change the max_cluster_count.

Thanks!
